internal/commands/app/create_inputs_test.go

```diff
@@ -372,111 +372,661 @@ func TestAppCreateInputsResolveDirectory(t *testing.T) {
+		datalakeNames := []string{"Datalake0", "Datalake1"}
+		datalakeServiceNames := []string{"mongodb-datalake", "another-data-source"}
+
+		ac := mock.AtlasClient{}
+		ac.DatalakesFn = func(groupID string) ([]atlas.Datalake, error) {
+			return []atlas.Datalake{
+				{Name: datalakeNames[0]},
+				{Name: datalakeNames[1]},
+			}, nil
+		}
+
+		for _, tc := range []struct {
+			description                  string
+			datalakeNames                []string
+			datalakeServiceNames         []string
+			procedure                    func(c *expect.Console)
+			autoConfirm                  bool
+			expectedDatalakeServiceNames []string
+		}{
+			{
+				description:                  "use data lake names provided",
+				datalakeNames:                datalakeNames,
+				datalakeServiceNames:         datalakeServiceNames,
+				procedure:                    func(c *expect.Console) {},
+				autoConfirm:                  false,
+				expectedDatalakeServiceNames: datalakeServiceNames,
+			},
+			{
+				description:   "prompt user if no data lake service names are provided",
+				datalakeNames: datalakeNames,
+				procedure: func(c *expect.Console) {
+					c.ExpectString("Enter a Service Name for Data Lake 'Datalake0'")
+					c.SendLine(datalakeServiceNames[0])
+					c.ExpectString("Enter a Service Name for Data Lake 'Datalake1'")
+					c.SendLine(datalakeServiceNames[1])
+					c.ExpectEOF()
+				},
+				autoConfirm:                  false,
+				expectedDatalakeServiceNames: datalakeServiceNames,
+			},
+			{
+				description:          "prompt user if any data lake service name is not provided",
+				datalakeNames:        datalakeNames,
+				datalakeServiceNames: []string{datalakeServiceNames[0]},
+				procedure: func(c *expect.Console) {
+					c.ExpectString("Enter a Service Name for Data Lake 'Datalake1'")
+					c.SendLine(datalakeServiceNames[1])
+					c.ExpectEOF()
+				},
+				autoConfirm:                  false,
+				expectedDatalakeServiceNames: datalakeServiceNames,
+			},
+			{
+				description:                  "default data lake service names to data lake names if not provided and auto confirm is set",
+				datalakeNames:                datalakeNames,
+				procedure:                    func(c *expect.Console) {},
+				autoConfirm:                  true,
+				expectedDatalakeServiceNames: datalakeNames,
+			},
+		} {
+			t.Run(tc.description, func(t *testing.T) {
+				console, _, ui, err := mock.NewVT10XConsoleWithOptions(mock.UIOptions{AutoConfirm: tc.autoConfirm})
+				assert.Nil(t, err)
+				defer console.Close()
+
+				doneCh := make(chan (struct{}))
+				go func() {
+					defer close(doneCh)
+					tc.procedure(console)
+				}()
+
+				inputs := createInputs{
+					newAppInputs:         newAppInputs{Name: "test-app"},
+					Datalakes:            tc.datalakeNames,
+					DatalakeServiceNames: tc.datalakeServiceNames,
+				}
+
+				ds, _, _ := inputs.resolveDatalakes(ui, ac, "123")
```

Comment:
missed actually setting `err` here ... here's the edge-case argument _in favor of_ `namedErr` (vs. `err`) 😬

Reply:
ah! good catch!